python/entities.py

```diff
@@ -14,6 +15,46 @@
 from jinja2 import utils
 
 
+class Property(object):
+    def __init__(self, remote_type, backref=None):
+        self.remote_type = remote_type
+        self.backref = backref
+
+
+class Meta(type):
+    def __new__(cls, name, bases, attrs):
+        scalar_properties = {}
+        recursive_properties = {}
+        attrs["_is_entity"] = True
+        for key, value in attrs.items():
+            if isinstance(value, Property):
+                if getattr(value.remote_type, '_is_entity', False):
```

Comment:
If I have to criticize something to show I have reviewed this pr, this is the only place that I can find I have some concerns.
1. is it possible to use isinstanceof here
2. is it possible to declare _is_entity directly in Entity
3, '_is_entity' is too general for a platform specific magic method, I suggest to use '_is_merchi_entity'.

Reply:
1. unfortunately no. `instanceof` checks the types, and at this stage, the type is not constructed yet... this code here is the part of the code responsible for creating an object and attaching a type (a class, specifically) to it.
2. the _ should already mark it as private, but ok

Reply:
you can see a few lines below, exactly where we create `new_cls`. so up here, the class isn't ready yet.